lib/GHCup/Types.hs

```diff
@@ -3,6 +3,7 @@
 {-# LANGUAGE BangPatterns      #-}
 {-# LANGUAGE DataKinds         #-}
 {-# LANGUAGE DeriveGeneric     #-}
+{-# LANGUAGE DeriveAnyClass    #-}
```

Comment:
I personally prefer avoiding `DeriveAnyClass` and just writing out the empty instance declarations, so that if later someone accidentally adds a class, that doesn't have an empty MINIMAL set, to the list in the `deriving` clause for `ArchiveResult`, we don't have to notice the compiler warning to avoid a runtime error.

Reply:
done